streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -671,7 +672,7 @@ void handleLostAll() {
         // just have an empty changelogOffsets map.
         for (final TaskId id : union(HashSet::new, lockedTaskDirectories, tasks.keySet())) {
             final Task task = tasks.get(id);
-            if (task != null) {
+            if (task != null && task.state() != State.CREATED) {
```

Comment:
Are there also other task states, we need to consider here except `CREATED`? What about `CLOSED` or `SUSPENDED`?

Reply:
Ahh, good catch: the task returns changelog offsets ultimately from the ProcessorStateManager's `stores` map, which is cleared during close. So we should fall through to the checkpoint file for `CLOSED` tasks as well (technically there's currently no way for a CLOSED task to be in the TaskManager's `tasks` map when the member sends a JoinGroup, since we remove it from the `tasks` immediately upon `close`, but that may not always be the case so better safe than sorry.

For `SUSPENDED` we should still rely on the task's. changelog offsets since it doesn't do the checkpoint or clearing out of `stores` during suspend. (Actually very little happens in `suspend` at the moment)